REFERENCE.md

```diff
@@ -0,0 +1,17 @@
+[//]: # (SPDX-License-Identifier: CC-BY-4.0)
+
+Relation to reference implementation
+====================================
+
+mlkem-native is a fork of the ML-KEM [reference implementation](https://github.com/pq-crystals/kyber).
+
+The following gives an overview of the major changes:
+
+- CBMC and debug annotations, and minor code restructurings or signature changes to facilitate the CBMC proofs. For example, `poly_add(x,a)` only comes in a destructive variant to avoid specifying aliasing constraints; `poly_rej_uniform` has an additional `offset` parameter indicating the position in the sampling buffer, to avoid passing shifted pointers).
+- Introduction of native backend implementations. With the exception of the native backend for `poly_rej_uniform()`, which may fail and fall back to the C implementation, those are drop-in replacements for the corresponding C functions and dispatched at compile-time.
+- Restructuring of files to separate level-specific from level-generic functionality. This is needed to enable a multi-level build of mlkem-native where level-generic code is shared between levels.
+- More pervasive use of value barriers to harden constant-time primitives. We are unaware of a practical issue with the reference implementation, but find our use of barriers more robust.
```

Comment:
You can stress that with these it's safe to turn on LTO. The reference implementation, relies on the constant time functions being and separate CUs and LTO being turned off.

Reply:
Good point